packages/request-state/src/runtime/clientState.ts

```diff
@@ -0,0 +1,39 @@
+import { parse } from 'devalue';
+
+const revivers: Record<string, (value: any) => any> = {
+	URL: (value) => new URL(value),
+	Date: (value) => new Date(value),
+	GlobalSymbol: (value) => Symbol.for(value),
+	WellKnownSymbol: (value) => Symbol[value as keyof typeof Symbol],
+};
+
+const loadState = (): Map<string, unknown> => {
+	const element = document.getElementById('it-astro-state');
+
+	if (element?.textContent) {
+		const state = parse(element.textContent, revivers);
+		element.remove();
```

Comment:
Could there be an interest to keep the element? I don't have a specific idea in mind, but maybe because of VT or something

Reply:
VT compatibility is a good point. I have to test with that, should be simple